app/scripts/metamask-controller.js

```diff
@@ -1609,7 +1614,7 @@ export default class MetamaskController extends EventEmitter {
 
     function updatePublicConfigStore(memState) {
       const { chainId } = networkController.store.getState().provider;
-      if (memState.network !== 'loading') {
+      if (memState.networkStatus === NetworkStatus.Available) {
```

Comment:
This looks different, but is effectively the same, as we have four states instead of two.